The A calculated that they need 135 more runs than they allow to win 95 wins which should allow to go to the world series. How can we validate that claim?

In [1]:
baseball = read.csv("baseball.csv")
str(baseball)

'data.frame':	1232 obs. of  15 variables:
 $ Team        : chr  "ARI" "ATL" "BAL" "BOS" ...
 $ League      : chr  "NL" "NL" "AL" "AL" ...
 $ Year        : int  2012 2012 2012 2012 2012 2012 2012 2012 2012 2012 ...
 $ RS          : int  734 700 712 734 613 748 669 667 758 726 ...
 $ RA          : int  688 600 705 806 759 676 588 845 890 670 ...
 $ W           : int  81 94 93 69 61 85 97 68 64 88 ...
 $ OBP         : num  0.328 0.32 0.311 0.315 0.302 0.318 0.315 0.324 0.33 0.335 ...
 $ SLG         : num  0.418 0.389 0.417 0.415 0.378 0.422 0.411 0.381 0.436 0.422 ...
 $ BA          : num  0.259 0.247 0.247 0.26 0.24 0.255 0.251 0.251 0.274 0.268 ...
 $ Playoffs    : int  0 1 1 0 0 0 1 0 0 1 ...
 $ RankSeason  : int  NA 4 5 NA NA NA 2 NA NA 6 ...
 $ RankPlayoffs: int  NA 5 4 NA NA NA 4 NA NA 2 ...
 $ G           : int  162 162 162 162 162 162 162 162 162 162 ...
 $ OOBP        : num  0.317 0.306 0.315 0.331 0.335 0.319 0.305 0.336 0.357 0.314 ...
 $ OSLG        : num  0.415 0.378 0.403 0.

In [3]:
moneyball = subset(baseball, Year<2002)
str(moneyball)

'data.frame':	902 obs. of  15 variables:
 $ Team        : chr  "ANA" "ARI" "ATL" "BAL" ...
 $ League      : chr  "AL" "NL" "NL" "AL" ...
 $ Year        : int  2001 2001 2001 2001 2001 2001 2001 2001 2001 2001 ...
 $ RS          : int  691 818 729 687 772 777 798 735 897 923 ...
 $ RA          : int  730 677 643 829 745 701 795 850 821 906 ...
 $ W           : int  75 92 88 63 82 88 83 66 91 73 ...
 $ OBP         : num  0.327 0.341 0.324 0.319 0.334 0.336 0.334 0.324 0.35 0.354 ...
 $ SLG         : num  0.405 0.442 0.412 0.38 0.439 0.43 0.451 0.419 0.458 0.483 ...
 $ BA          : num  0.261 0.267 0.26 0.248 0.266 0.261 0.268 0.262 0.278 0.292 ...
 $ Playoffs    : int  0 1 1 0 0 0 0 0 1 0 ...
 $ RankSeason  : int  NA 5 7 NA NA NA NA NA 6 NA ...
 $ RankPlayoffs: int  NA 1 3 NA NA NA NA NA 4 NA ...
 $ G           : int  162 162 162 162 161 162 162 162 162 162 ...
 $ OOBP        : num  0.331 0.311 0.314 0.337 0.329 0.321 0.334 0.341 0.341 0.35 ...
 $ OSLG        : num  0.412 0.404 0.384 0.

In [5]:
moneyball$RD = moneyball$RS - moneyball$RA
str(moneyball)

'data.frame':	902 obs. of  16 variables:
 $ Team        : chr  "ANA" "ARI" "ATL" "BAL" ...
 $ League      : chr  "AL" "NL" "NL" "AL" ...
 $ Year        : int  2001 2001 2001 2001 2001 2001 2001 2001 2001 2001 ...
 $ RS          : int  691 818 729 687 772 777 798 735 897 923 ...
 $ RA          : int  730 677 643 829 745 701 795 850 821 906 ...
 $ W           : int  75 92 88 63 82 88 83 66 91 73 ...
 $ OBP         : num  0.327 0.341 0.324 0.319 0.334 0.336 0.334 0.324 0.35 0.354 ...
 $ SLG         : num  0.405 0.442 0.412 0.38 0.439 0.43 0.451 0.419 0.458 0.483 ...
 $ BA          : num  0.261 0.267 0.26 0.248 0.266 0.261 0.268 0.262 0.278 0.292 ...
 $ Playoffs    : int  0 1 1 0 0 0 0 0 1 0 ...
 $ RankSeason  : int  NA 5 7 NA NA NA NA NA 6 NA ...
 $ RankPlayoffs: int  NA 1 3 NA NA NA NA NA 4 NA ...
 $ G           : int  162 162 162 162 161 162 162 162 162 162 ...
 $ OOBP        : num  0.331 0.311 0.314 0.337 0.329 0.321 0.334 0.341 0.341 0.35 ...
 $ OSLG        : num  0.412 0.404 0.384 0.

In [7]:
WinsReg = lm(W ~ RD, data=moneyball)
summary(WinsReg)


Call:
lm(formula = W ~ RD, data = moneyball)

Residuals:
     Min       1Q   Median       3Q      Max 
-14.2662  -2.6509   0.1234   2.9364  11.6570 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 80.881375   0.131157  616.67   <2e-16 ***
RD           0.105766   0.001297   81.55   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 3.939 on 900 degrees of freedom
Multiple R-squared:  0.8808,	Adjusted R-squared:  0.8807 
F-statistic:  6651 on 1 and 900 DF,  p-value: < 2.2e-16


---

80.881375 Intercept or $b$ from $y=mx=b$
RD

$w=80.8814 + 0.1058(RD)$

We want wins to be greater than 95.

$W\geq95$

$80.8814+0.105(RD)\geq95$

$RD\geq (95-80.8814)/0.1058=133.4 $

---


* How does a team score more runs?
    * On-Base Percentage(OBP)
        * Percentage of time a player gets on base (including walks)
    * Slugging Percentage (SLG)
        * How far a player gets around the bases on his turn (measures power)
    * Most teams focused on Batting Average(BA)
        
    * The A's claimed that
        * On-Base Percentage was the most important
        * Slugging Percentage was important
        * Batter Average was overvalued
        
## * Can we use linear regression to verify which baseball stats are more important to predict runs?


In [8]:
str(moneyball)

'data.frame':	902 obs. of  16 variables:
 $ Team        : chr  "ANA" "ARI" "ATL" "BAL" ...
 $ League      : chr  "AL" "NL" "NL" "AL" ...
 $ Year        : int  2001 2001 2001 2001 2001 2001 2001 2001 2001 2001 ...
 $ RS          : int  691 818 729 687 772 777 798 735 897 923 ...
 $ RA          : int  730 677 643 829 745 701 795 850 821 906 ...
 $ W           : int  75 92 88 63 82 88 83 66 91 73 ...
 $ OBP         : num  0.327 0.341 0.324 0.319 0.334 0.336 0.334 0.324 0.35 0.354 ...
 $ SLG         : num  0.405 0.442 0.412 0.38 0.439 0.43 0.451 0.419 0.458 0.483 ...
 $ BA          : num  0.261 0.267 0.26 0.248 0.266 0.261 0.268 0.262 0.278 0.292 ...
 $ Playoffs    : int  0 1 1 0 0 0 0 0 1 0 ...
 $ RankSeason  : int  NA 5 7 NA NA NA NA NA 6 NA ...
 $ RankPlayoffs: int  NA 1 3 NA NA NA NA NA 4 NA ...
 $ G           : int  162 162 162 162 161 162 162 162 162 162 ...
 $ OOBP        : num  0.331 0.311 0.314 0.337 0.329 0.321 0.334 0.341 0.341 0.35 ...
 $ OSLG        : num  0.412 0.404 0.384 0.

In [9]:
RunsReg = lm(RS~OBP + SLG +BA, data=moneyball)
summary(RunsReg)


Call:
lm(formula = RS ~ OBP + SLG + BA, data = moneyball)

Residuals:
    Min      1Q  Median      3Q     Max 
-70.941 -17.247  -0.621  16.754  90.998 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  -788.46      19.70 -40.029  < 2e-16 ***
OBP          2917.42     110.47  26.410  < 2e-16 ***
SLG          1637.93      45.99  35.612  < 2e-16 ***
BA           -368.97     130.58  -2.826  0.00482 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 24.69 on 898 degrees of freedom
Multiple R-squared:  0.9302,	Adjusted R-squared:   0.93 
F-statistic:  3989 on 3 and 898 DF,  p-value: < 2.2e-16


Coe is negative. A team with a higher batting average will score fewer runs. (Counter intuitive)

This is a case of multicollinearity are highly coorelated.

Remove BA


In [10]:
RunsReg = lm(RS~OBP + SLG, data=moneyball)
summary(RunsReg)


Call:
lm(formula = RS ~ OBP + SLG, data = moneyball)

Residuals:
    Min      1Q  Median      3Q     Max 
-70.838 -17.174  -1.108  16.770  90.036 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  -804.63      18.92  -42.53   <2e-16 ***
OBP          2737.77      90.68   30.19   <2e-16 ***
SLG          1584.91      42.16   37.60   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 24.79 on 899 degrees of freedom
Multiple R-squared:  0.9296,	Adjusted R-squared:  0.9294 
F-statistic:  5934 on 2 and 899 DF,  p-value: < 2.2e-16


coe for OBP is higher than coe SLG

OBP is worth more than SLG

BA is over valued and OBP is the most important

---

## Allowing Runs
* We can use pitching statistics to predict runs allowed
    * Opponents On-Base Percentage (OOBP)
    * Opponents Slugging Percentage (OSLG)
* We get the linear regression model

    $Runs Allowed = -837.38 + 2913.60 (OOBP) + 1514(OSLG)$

* $R^2 = 0.91 $

* Both variables significant

### If a baseball team's OBP is 0.311 and SLG is 0.405, how many runs do we expect the team to score?

#### Using the linear regression model constructed during the lecture (the one that uses OBP and SLG as independent variables), enter the number of runs we expect the team to score:

---

### If a baseball team's opponents OBP (OOBP) is 0.297 and oppenents SLG (OSLG) is 0.370, how many runs do we expect the team to allow?

#### Using the linear regression model discussed during the lecture (the one on the last slide of the previous video), enter the number of runs we expect the team to allow:



---
### Predicting Runs and WIns

* Can we predict how many games the 2002 Oakland A's will win using our models?
* The models for runs use team statistics
* Each year, a baseball team is different.
* We need to estimate the new team statistics using past player performance
* Assumes few injuries
* We can estimate the team statistics for 2002 by using the 2001 player statistics

### Predicting Runs Scored 
* At the beginning of the 2002 season, the Oakland A's had 24 batters on their roster
* Using the 2001 regular season statistics for these players
    * Team OBP is 0.339
    * Team SLG is 0.430
* Our regression equation was 

$RS = -804.63 + 2737.77(OBP)+1584.91(SLG)$

* Our 2002 prediction for the A's is

$RS = -804.63 + 2737.77(0.339)+1584.91(0.430)=805$


### Predicting Runs Allowed
* At the beginning of the 2002 season, the Okland A's had 17 picters on their roster
* Using the 2001 regular season statistics for these players
    * Team OOBP is 3.07
    * Team OSLG is 0.373
* Our regression equation was
    * RA = -837.38 + 2913.60(OOBP) + 1514.29(OSLG) 
* Our 2002 prediction for the A’s is 
    * RA = -837.38 + 2913.60(0.307) + 1514.29 (0.373) = 622 
    
### Predicting Wins
* Our regression equation to predict wins was
    * Wins = 80.8814 + 0.1058(RS – RA) 
 
* We predicted 
* RS = 805 
* RA = 622 
 
* So our prediction for wins is
    * Wins = 80.8814 + 0.1058(805 – 622) = 100 

The Oakland A’s
• Paul DePodesta used a similar approach to make predictions
• Predictions closely match actual performance
Our
Prediction
Paul’s
Prediction Actual
Runs Scored 805 800 – 820 800 

The Oakland A’s
• Paul DePodesta used a similar approach to make predictions
• Predictions closely match actual performance
Our
Prediction
Paul’s
Prediction Actual
Runs Scored 805 800 – 820 800
Runs Allowed 622 650 – 670 653 

While Moneyball made the use of analytics in sports very popular, baseball is not the only sport for which analytics is used. Analytics is currently used in almost every single sport, including basketball, soccer, cricket, and hockey.

Basketball: The study of analytics in basketball, called APBRmetrics, is very popular. There have been many books written in this area, including "Pro Basketball Forecast" by John Hollinger and "Basketball on Paper" by Dean Oliver. There are also several websites dedicated to the study of basketball analytics, including [82games.com](http://www.82games.com). We'll talk more about basketball during recitation.

Soccer: The soccer analytics community is currently growing, and new data is constantly being collected. Many argue that it is much harder to apply analytics to soccer, but there are several books and websites on the topic. Check out "The Numbers Game: Why Everything You Know about Football is Wrong" by Chris Anderson and David Sally, as well as the websites [socceranalysts.com](http://www.socceranalysts.com) and [soccermetrics.net](http://www.soccermetrics.net).

Cricket: There are several websites dedicated to building models for evaluating player performance in cricket. Check out [cricmetric.com](http://www.cricmetric.com) and [impactindexcricket.com](http://www.impactindexcricket.com).

Hockey: Analytics are used in hockey to track player performance and to better shape the composition of teams. Check out the websites [hockeyanalytics.com](http://www.hockeyanalytics.com) and [lighthousehockey.com](http://www.lighthousehockey.com).

Submit